### ISSUE TITLE
Adding new worker node to cluster fails to register in Rancher v2.4.2 and v2.4.3-rc1

### ISSUE THREAD

**Issue by zbialik** [rancher/rancher#26776]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

I was looking forward to the upgrade for the new LDAP features but I officially cannot upgrade until we resolve this error. Essentially can't perform cluster-autoscaling, nor patching of infrastructure.

**Steps to reproduce (least amount of steps as possible):**
Initialize rancher v2.4.2
Initialize kubernetes cluster (from custom nodes) and configure amazon as cloud provider
Bootup a new ec2 instance and attempt to bootstrap with docker command provided by rancher

**Result:**
Worker node gets stuck in Registering state.

**Other details that may be helpful:**
I thought this could be related to [23254 ](https://github.com/rancher/rancher/issues/23254) but after testing with rancher v2.4.3-rc1, I hit the same error. Not sure why. Maybe v2.4.3-rc1 doesn't have the fix yet?

kubelet container is never officially created.

the containers running are shown below:
```
[root@ip-XX-XX-XX-XX ~]# docker ps
CONTAINER ID        IMAGE                                                                                  COMMAND                  CREATED             STATUS              PORTS               NAMES
931c16d2236d        docker-registry.example.com:5000/rancher/rke-tools:v0.1.56            "nginx-proxy CP_HO..."   5 minutes ago       Up 4 minutes                            nginx-proxy
af4c220fdd5b        docker-registry.example.com:5000/rancher/rancher-agent:v2.4.3-rc1     "run.sh --server h..."   5 minutes ago       Up 5 minutes                            stupefied_wiles
f594377940f9        docker-registry.example.com:5000/rancher/hyperkube:v1.17.5-rancher1   "/opt/rke-tools/en..."   5 minutes ago       Up 5 minutes                            kube-proxy
cc08bc6af25b        docker-registry.example.com:5000/rancher/rancher-agent:v2.4.3-rc1     "run.sh -- share-r..."   6 minutes ago       Up 5 minutes                            share-mnt
6ad5a313d951        docker-registry.example.com:5000/rancher/rancher-agent:v2.4.3-rc1     "run.sh --server h..."   6 minutes ago       Up 6 minutes                            pedantic_curran
```

the initial rancher-agent container (container named pedantic_curran) has these logs repeated infinitely:

```
time="2020-04-23T15:20:30Z" level=info msg="Connecting to wss://rancher.example.com/v3/connect with token 65z4p9bjbmffznjx2s9c8gblacbkajiurbeiivuhaiurv7xm9g7rd"
time="2020-04-23T15:20:30Z" level=info msg="Connecting to proxy" url="wss://rancher.example.com/v3/connect"
time="2020-04-23T15:20:30Z" level=error msg="Remotedialer proxy error" error="Error response from daemon: invalid bind mount spec \"::rw,rslave,z\": invalid volume specification: '::rw,rslave,z'"
time="2020-04-23T15:20:40Z" level=info msg="Connecting to wss://rancher.example.com/v3/connect with token 65z4p9bjbmffznjx2s9c8gblacbkajiurbeiivuhaiurv7xm9g7rd"
time="2020-04-23T15:20:40Z" level=info msg="Connecting to proxy" url="wss://rancher.example.com/v3/connect"
time="2020-04-23T15:20:40Z" level=error msg="Remotedialer proxy error" error="Error response from daemon: invalid bind mount spec \"::rw,rslave,z\": invalid volume specification: '::rw,rslave,z'"
```

the kube-proxy has logs indicating unauthorized access to api:

```
I0423 15:17:18.157255   11241 server.go:571] Version: v1.17.5
I0423 15:17:18.158026   11241 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I0423 15:17:18.158061   11241 conntrack.go:52] Setting nf_conntrack_max to 131072
I0423 15:17:18.158134   11241 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I0423 15:17:18.158185   11241 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I0423 15:17:18.158766   11241 config.go:313] Starting service config controller
I0423 15:17:18.158781   11241 shared_informer.go:197] Waiting for caches to sync for service config
I0423 15:17:18.158800   11241 config.go:131] Starting endpoints config controller
I0423 15:17:18.158822   11241 shared_informer.go:197] Waiting for caches to sync for endpoints config
E0423 15:17:18.163444   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:18.163526   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:18.163572   11241 event.go:272] Unable to write event: 'Post https://127.0.0.1:6443/api/v1/namespaces/default/events: dial tcp 127.0.0.1:6443: connect: connection refused' (may retry after sleeping)
E0423 15:17:19.164068   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:19.165109   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:20.164736   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:20.165656   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:21.165382   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:21.166352   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:22.165981   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:22.167025   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:23.166611   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:23.168416   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Service: Get https://127.0.0.1:6443/api/v1/services?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
E0423 15:17:24.167238   11241 reflector.go:153] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.Endpoints: Get https://127.0.0.1:6443/api/v1/endpoints?labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&limit=500&resourceVersion=0: dial tcp 127.0.0.1:6443: connect: connection refused
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.4.2 and also v2.4.3-rc1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): t2.large
- Kubernetes version (use `kubectl version`):

```
 v1.17.4-rancher1-3 and also v1.17.5-rancher1-1 (when using rancher v2.4.3-rc1)
```

- Docker version (use `docker version`):

```
1.13.1
```

**Comment by zbialik** [comment]
Closing as I just verified that the commit, [00ff159](https://github.com/rancher/rancher/commit/00ff159769298db1920dd61ed246eaf5046544ef), which resolves [this issue](https://github.com/rancher/rancher/issues/23254), was committed after v2.4.3-rc1 was released.

I will wait for the next release to verify that the commit fixes my issues.

**Comment by zbialik** [comment]
reopening as im not sure this is actually fixed..

**Comment by ekristen** [comment]
I can confirm this problem just happened on one of my clusters.

**Comment by zbialik** [comment]
@ekristen I noticed the other [issue](https://github.com/rancher/rancher/issues/26790) post tracking this problem recently closed after someone tested using rancher:v2.4.3-rc3. I am going to test using that version and see if this will be resolved by v2.4.3

**Comment by zbialik** [comment]
I have verified that v2.4.3-rc3 fixes this bug. Closing.
